resteasy-spring-feature-pack/common/src/main/resources/layers/standalone/resteasy-spring-web/layer-spec.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:2.0" name="resteasy-spring-web">
+    <props>
+        <prop name="org.wildfly.rule.xml-path" value="/WEB-INF/web.xml,/web-app/listener/listener-class,org.jboss.resteasy.plugins.server.servlet.ResteasyBootstrap"/>
```

Comment:
This is the same rule as in resteasy-spring layer, does it means that the 2 layers should get provisioned all together if the servlet is found? What is the use case associated to each layer? If these 2 need to get provisioned together, then perhaps we need a dependency from resteasy-spring-web to resteasy-spring?

Reply:
That's a good point. I don't really know the answer to that. @liweinan do you know? 

I guess the point would be if they both have the same rule we likely shouldn't separate them. I just don't know enough about Spring to know what's required and what's not. I will say it doesn't make sense to keep both of these properties the same though.

My instinct would tell me for `resteasy-spring` we should just look for a Spring class. That said, I don't know what the difference between `resteasy-spring` and `resteasy-spring-web` really is. Like what does the `resteasy-spring-web` add that?

Reply:
@jamezp , thank-you, yes that is what we need to solve to be able to define the right set of layers. @liweinan , it would be great if we could have examples of each "usages". Thank-you!

Reply:
@jamezp @jfdenise The `resteasy-spring-web` subproject allows the users to use the `spring-web` annotations instead of the `JAX-RS` annotations to annotate there resources. In the testsuite it contains some usage examples:

- https://github.com/resteasy/resteasy-spring/tree/main/testsuite/integration-tests-spring-web

For example here is a sample resource:

- https://github.com/resteasy/resteasy-spring/blob/main/testsuite/integration-tests-spring-web/deployment/src/test/java/org/jboss/resteasy/test/spring/web/deployment/resource/ResponseEntityController.java

With my understanding it's a "thin" layer that supports the `spring-web` annotations in RESTEasy.

So the configuration in `web.xml` is different from a `resteasy-spring` based project which uses the full Springframework features:

- https://github.com/resteasy/resteasy-spring/blob/main/testsuite/integration-tests-spring-web/deployment/src/test/resources/org/jboss/resteasy/test/spring/web/deployment/web.xml#L11

There is document describing its usage:

- https://docs.jboss.org/resteasy/docs/6.2.6.Final/userguide/html_single/#SpringWebAnnotation

So I guess the trigger rule here needs to be adjusted accordingly.

Reply:
According to the doc, the trigger rule should be:

> RESTEasy does not auto-scan for @RestController annotated classes, so all @RestController annotated classes need to be declared in the application's web.xml file as shown below.
> 
> 
>  <web-app>
>    <display-name>RESTEasy application using Spring REST annotations</display-name>
> 
>     <context-param>
>         <param-name>resteasy.scanned.resource.classes.with.builder</param-name>
>         <param-value>org.jboss.resteasy.spi.metadata.SpringResourceBuilder:org.example.Controller1,org.example.Controller2</param-value>
>     </context-param>
>    ...
> </web-app>
>

Reply:
@jamezp @liweinan so the rule to detect annotated classes should be (if all anottations are location the same package): 
```<prop name="org.wildfly.rule.annotations" value="org.springframework.web.bind.annotation"/>```
With this rule inside resteasy-spring-web layer, this layer should get picked-up.

Reply:
Yes, that makes sense to me. I'll get this updated.

Reply:
@jamezp Thanks for the update!

Reply:
Updated and I changed the `web.xml` rule to look for to look for `org.jboss.resteasy.plugins.spring.SpringContextLoaderListener` as that is part of RESTEasy Spring and `ResteasyBootstrap` is part of resteasy-core and may be defined even if Spring is not used.

Reply:
The changes looks good to me. Guess we can adjust it in the future if any rules need to be adjusted. Thanks for working on this James!